sys/slot_util/slot_util.c

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (C) 2017 Kaspar Schleiser <kaspar@schleiser.de>
+ *                    Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     sys_slot_util
+ * @{
+ *
+ * @file
+ * @brief       Slot management functions
+ *
+ * @author      Kaspar Schleiser <kaspar@schleiser.de>
+ * @author      Francisco Acosta <francisco.acosta@inria.fr>
+ *
+ * @}
+ */
+#include <string.h>
+
+#include "cpu.h"
+#include "slot_util.h"
+#include "riot_hdr.h"
+
+/*
+ * Store the start addresses of each slot.
+ * Take into account that CPU_FLASH_BASE represents the starting
+ * address of the bootloader, thus the header is located after the
+ * space reserved to the bootloader.
+ */
+const riot_hdr_t * const slot_util_slots[] = {
+    (riot_hdr_t*)(CPU_FLASH_BASE + SLOT0_OFFSET),   /* First slot address -> firmware image */
+};
+
+/* Calculate the number of slots */
+const unsigned slot_util_num_slots = sizeof(slot_util_slots) / sizeof(uint32_t);
```

Comment:
Formally shouldn't it be:
` const unsigned slot_util_num_slots = sizeof(slot_util_slots) / sizeof(riot_hdr_t *); `

Reply:
+1

It is also common to use directly a variable item so here `slot_util_slots[0]`.

`#define ARRAY_LEN(array) (sizeof(array)/sizeof(array[0]))`

Reply:
It was a leftover, I fixed it.